hapi-fhir-storage/src/main/java/ca/uhn/fhir/jpa/dao/tx/HapiTransactionService.java

```diff
@@ -74,18 +73,13 @@ public class HapiTransactionService implements IHapiTransactionService {
 	protected PlatformTransactionManager myTransactionManager;
 	@Autowired
 	protected IRequestPartitionHelperSvc myRequestPartitionHelperSvc;
-	private boolean myCustomIsolationSupported;
+	private Boolean myCustomIsolationSupported;
```

Comment:
```suggestion
	private volatile Boolean myCustomIsolationSupported;
```

If you're potentially setting this field from multiple threads, it needs to be volatile.

Reply:
nice catch